grails-webhooks/grails-app/services/webhooks/WebhookService.groovy

```diff
@@ -159,9 +159,17 @@ class WebhookService {
             hook.uuid = UUID.randomUUID().toString()
         }
         hook.uuid = hookData.uuid ?: hook.uuid
-        int countByNameInProject = Webhook.countByNameAndProject(hookData.name,hookData.project)
-        if(countByNameInProject > 0) {
-            return [err: " A Webhook by that name already exists in this project"]
+        // If there's an import, we validate that the name of the WH is not duplicated in the project.
+        int countByNameInProject = Webhook.countByNameAndProject(hookData.name, hookData.project)
+        if (countByNameInProject > 0) {
+            // If we find during the import a WH with the same name, we check for duplicated UUID.
+            int countByUuidInProject = Webhook.countByUuidAndProject(hookData.uuid, hookData.project)
+            // If we find the UUID, it means that is a different webhook with duplicated name, so we prevent the import
+            if( countByUuidInProject == 0 ) return [err: "A Webhook by that name already exists in project"]
+        }else{
+            // If we don't find the WH by name in the project, we check if the WH exists in other projects to prevent errors
+            int countByUuidInDb = Webhook.countByUuid(hookData.uuid)
```

Comment:
This count prevent renaming the job, need to exclude current project

Reply:
Looking at the previous code, the current behavior is to allow having the same uuid on multiple projects, the uuid only is verified if exists on the current project so this need to be removed